pkg/retrieval/retrieval.go

```diff
@@ -248,8 +202,23 @@ func (s *Service) RetrieveChunk(ctx context.Context, addr swarm.Address, origin
 	return v.(swarm.Chunk), nil
 }
 
-func (s *Service) retrieveChunk(ctx context.Context, addr swarm.Address, sp *skipPeers, originated bool) (chunk swarm.Chunk, peer swarm.Address, requested bool, err error) {
-	startTimer := time.Now()
+func (s *Service) retrieveChunk(ctx context.Context, resultChan chan retrievalResult, doneChan chan struct{}, addr swarm.Address, sp *skipPeers, originated bool) {
+
+	var (
+		peer       swarm.Address
+		err        error
+		chunk      swarm.Chunk
+		retrieved  bool
+		startTimer = time.Now()
+	)
+
+	defer func() {
+		select {
+		case resultChan <- retrievalResult{peer: peer, err: err, retrieved: retrieved, chunk: chunk}:
+		case <-doneChan:
+		}
```

Comment:
maybe also add a similar duplicate chunk metric here, similar to what we do in pushsync